nep-Y.mediawiki

```diff
@@ -0,0 +1,98 @@
+<pre>
+  NEP: <to be assigned>
+  Title: Smart contract transfer callback for non-fungible tokens
+  Author: Erik Zhang <erik@neo.org>, Roman Khimov <roman@nspcc.ru>
+  Type: Informational
+  Status: Draft
+  Created: 2024-03-03
+  Requires: 11
+</pre>
+
+==Abstract==
+
+This proposal describes the callback method used by Neo smart contracts to
+process transfers of non-fungible [[nep-11.mediawiki|NEP-11]] tokens to their addresses.
+
+==Motivation==
+
+While the original [[nep-11.mediawiki|NEP-11]] already defines this method, it
+lacks some details relevant specifically to the receiver contract and it doesn't
+allow for easy reference. This standard can be used in <code>supportedstandards</code>
+section of [[nep-15.mediawiki|NEP-15 manifest]] or in any other situation to
+refer to contracts able to receive [[nep-11.mediawiki|NEP-11]] tokens (but not
+necessarily being [[nep-11.mediawiki|NEP-11]] themselves). Typical use cases
+for this functionality are: depositing/staking (with contract locking and
+managing tokens according to some rules) or exchange/minting (contract accepting
+transfer to itself, but returning some different token to the sender).
+
+==Specification==
+
+Smart contracts that need to receive and own [[nep-11.mediawiki|NEP-11]] tokens
+MUST implement <code>onNEP11Payment</code> method with the following signature:
+
+<pre>
+{
+  "name": "onNEP11Payment",
+  "parameters": [
+    {
+      "name": "from",
+      "type": "Hash160"
+    },
+    {
+      "name": "amount",
+      "type": "Integer"
+    },
+    {
+      "name": "tokenId",
+      "type": "ByteString"
+    },
+    {
+      "name": "data",
+      "type": "Any"
+    }
+  ],
+  "returntype": "Void"
+}
+</pre>
+
+A contract that doesn't implement this method will not be able to receive
+[[nep-11.mediawiki|NEP-11]] tokens (transfers to its address will fail). This
```

Comment:
This is the current code behaviour. Are we going to change it? I.e. what happens if a contract has OnNEPXXPayment method but does not have the corresponding NEPY declared in its manifest?

Reply:
Why should it be changed? The behavior should stay the same. These NEPs are _informational_.

Reply:
> This is the current code behaviour. Are we going to change it? I.e. what happens if a contract has OnNEPXXPayment method but does not have the corresponding NEPY declared in its manifest?

@AnnaShaleva having this method implemented is sufficient to make a contract payable, we can not change this logic. NEP is just for declaration, like NEP11 and NEP17, it does not matter if a contract declares itself as support NEP11 or NEP17 or not.